### ISSUE TITLE
NullPointerException when loading System props in Teiid embedded server

### ISSUE THREAD

**Issue by gamefundas** [stagemonitor/stagemonitor#170]
This is perhaps a duplicate of #140 
Wondering if it was resolved completely as we still seem to get a NPE error when Stagemonitor is configured in our embedded Teiid Server.

```
Caused by: java.lang.RuntimeException: unable to load properties from jar:file:/P:/Projects/Notification_Frmwrk/NF_wrkspc/data_service_beta_workspace/app/target/data-service.jar!/lib/narayana-jta-5.2.13.Final.jar!/jbossts-properties.xml
	at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:107)
	at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.initDefaultProperties(AbstractPropertiesFactory.java:195)
	at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getDefaultProperties(AbstractPropertiesFactory.java:62)
	at com.arjuna.common.util.propertyservice.PropertiesFactory.getDefaultProperties(PropertiesFactory.java:37)
	at com.arjuna.common.internal.util.propertyservice.BeanPopulator.getNamedInstance(BeanPopulator.java:86)
	... 67 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.util.Hashtable.put(Unknown Source)
	at java.util.Properties.setProperty(Unknown Source)
	at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.applySystemProperties(AbstractPropertiesFactory.java:124)
	at com.arjuna.common.util.propertyservice.AbstractPropertiesFactory.getPropertiesFromFile(AbstractPropertiesFactory.java:104)
	... 71 common frames omitted
```

**Comment by felixbarny** [comment]
Actually, I can't find any place within stagemonitor which puts a non string value into the system properties anymore. Which version of stagemonitor are you using? Could you find out the property key which is associated with a non string value? That would be really helpful!

**Comment by gamefundas** [comment]
The version of SM is 0.25.0 and the observation is that when SM instruments the code the property loading mechanism of libraries like narayanajta which contains the property file inside it fails. 

The NPE is raised when the libraries fails to find its own properties bundled in the jar.

**Comment by felixbarny** [comment]
Could you debug `AbstractPropertiesFactory.java:124` to see which property key is causing the problems?

**Comment by get2parag** [comment]
Hi,
I'm also working on the same issue. The property in question is stagemonitor.requestmonitor.deleteRequestTracesAfterDays. The value which was already set to this key : -1.

**Comment by felixbarny** [comment]
Do you set this value via the command line like `java -jar your-app.jar -Dstagemonitor.requestmonitor.deleteRequestTracesAfterDays=-1`? Is the value `-1` of type `String` or something else? How can I reproduce the error?

**Comment by felixbarny** [comment]
Normally, stagemonitor does not write properties like `stagemonitor.requestmonitor.deleteRequestTracesAfterDays` to the system properties. I'm quite puzzled how this can happen. So I'd be really grateful for an explanation on how to reproduce this error.

**Comment by gamefundas** [comment]
We debugged the issue and found that this is because of the way the third party library was converting all properties to System properties (including SM one which ideally it shouldn't be looking at but is a bug in narayana jta) and the assumption was all properties were String. In our SM configuration we had used Integers and Boolean for some of them and this was in conflict. 

Resolved this by configuring all SM properties as String in our Spring boot module to solve the issue.
Sorry for the mix up, should have debugged this earlier.

We should be able to close this one, as a non issue.
